LCM/scripts/PerformInventory.py

```diff
@@ -99,8 +101,8 @@ def usage():
     sys.exit(1)
 
 # Acquire inventory file lock
-filehandle = os.open(dsc_sysconfdir + "/inventory_lock", os.O_WRONLY | os.O_CREAT , 0o644)
-inventory_lock = os.fdopen(filehandle, 'w')
+inventory_lock = open(inventorylock_path, "w+")
+os.chmod(inventorylock_path , stat.S_IRUSR | stat.S_IWUSR | stat.S_IRGRP | stat.S_IROTH)
```

Comment:
Here we are changing the permissions right away. I don't see any time gap between file creation and doing chmod. The problem is when we were creating the file with those permissions before it was only recreating the file if it does not exist and if file already has ww it was not overwriting the permissions on upgrade. This is not what API said it should do but I suspect since we put locks on this file it was not recreating the file. Now this always creates the new file and then sets the permission no matter if it exists before or not. This is the behavior we want.